spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestUpdate.java

```diff
@@ -1455,6 +1457,31 @@ public void testUpdateToWapBranchWithTableBranchIdentifier() {
                         branch)));
   }
 
+  @Test
+  public void subExpressionEliminationInCodegen() {
```

Comment:
should we now run TestUpdate both with / without Codegen ? as if at the moment there is a fallback is there is an issue that it silently fallbacks to interpreted mode which we don't intend, thoughts ?

Reply:
I thought about disabling the fallback to cut the time. Maybe, we can further parameterize this suite. Let me see.